arch/arm64/src/common/arm64_registerdump.c

```diff
@@ -50,7 +50,7 @@
 uintptr_t up_getusrsp(void *regs)
 {
   struct regs_context *ptr = regs;
-  return ptr->regs[REG_X13];
+  return ptr->sp_elx;
```

Comment:
sp_elx is the kernel sp

Reply:
kernel build sp_el0 is user stack,
flat build sp_elx is user stack